cpp/test/sg/rf_test.cu

```diff
@@ -40,370 +101,357 @@ struct RfInputs {
+                                                                           bootstrap,
+                                                                           n_bins,
+                                                                           min_samples_leaf,
+                                                                           min_samples_split,
+                                                                           min_impurity_decrease,
+                                                                           n_streams,
+                                                                           split_criterion,
+                                                                           seed,
+                                                                           n_labels,
+                                                                           double_precision)));
+
+struct QuantileTestParameters {
+  int n_rows;
+  int n_bins;
+  uint64_t seed;
+};
 
-INSTANTIATE_TEST_CASE_P(RfClassifierTests, RfClassifierTestF, ::testing::ValuesIn(inputsf2_clf));
+template <typename T>
+class RFQuantileTest : public ::testing::TestWithParam<QuantileTestParameters> {
+ public:
+  void SetUp() override
+  {
+    auto params = ::testing::TestWithParam<QuantileTestParameters>::GetParam();
+
+    thrust::device_vector<T> data(params.n_rows);
+    thrust::device_vector<T> quantiles(params.n_bins);
+    thrust::device_vector<int> histogram(params.n_bins);
+    thrust::host_vector<int> h_histogram(params.n_bins);
+
+    raft::random::Rng r(8);
+    r.normal(data.data().get(), data.size(), T(0.0), T(2.0), nullptr);
+    raft::handle_t handle;
+    DT::computeQuantiles(quantiles.data().get(),
+                         params.n_bins,
+                         data.data().get(),
+                         params.n_rows,
+                         1,
+                         handle.get_device_allocator(),
+                         nullptr);
+
+    auto d_quantiles = quantiles.data();
+    auto d_histogram = histogram.data().get();
+    thrust::for_each(data.begin(), data.end(), [=] __device__(T x) {
+      for (int j = 0; j < params.n_bins; j++) {
+        if (x <= d_quantiles[j]) {
+          atomicAdd(&d_histogram[j], 1);
+          break;
+        }
+      }
+    });
+
+    h_histogram           = histogram;
+    int max_items_per_bin = raft::ceildiv(params.n_rows, params.n_bins);
+    int min_items_per_bin = max_items_per_bin - 1;
+    int total_items       = 0;
+    for (int b = 0; b < params.n_bins; b++) {
+      ASSERT_TRUE(h_histogram[b] == max_items_per_bin || h_histogram[b] == min_items_per_bin)
+        << "No. samples in bin[" << b << "] = " << h_histogram[b] << " Expected "
+        << max_items_per_bin << " or " << min_items_per_bin << std::endl;
+      total_items += h_histogram[b];
+    }
+    ASSERT_EQ(params.n_rows, total_items)
+      << "Some samples from dataset are either missed of double counted in "
```

Comment:
The string constant can fit on single line now, as the line width is increased from 80 to 100 column with latest clan-formatting rules.